### PR TITLE
Add tutorial notebooks 

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Long time awaited PR: Tutorials on how to use PyFstat to do different things!

For now:

- [x] How to generate Gaussian noise.
- [x] How to inject signals.

To do:
(+ @mj-will comments)

- [ ] Make sure binder works.
- [x] Likewise for documentation --> Should modify `examples.rst` and include it as an entry.
- The section links in both notebooks don't seem to work for me, [...] but probably worth double-checking they work.
     - [x] Works on local jupyter.
     - [ ] Works on binder.
- [x] Might be worth starting explicitly that these notebooks need at least v1.* [...] in case conda/pip installs an older version

### CODE REVIEW

**Comment by Rodrigo-Tenorio** [comment]
@mj-will tutorial 0 is go for comments :rocket:

**Comment by mj-will** [comment]
So generally I think it looks really good but I do have a couple of comments

*  The section links in both notebooks don't seem to work for me, not sure why (it could be VSCode) but probably worth double-checking they work
*  There are some unused imports in the signal notebook
* Might be worth starting explicitly that these notebooks need at least v1.* (which ever the latest version is) just in case conda/pip installs an older version

**Comment by Rodrigo-Tenorio** [comment]
@dbkeitel I'm having problems with binder, any problems in merging and then submitting PR to fix it if it fails in master?

File: examples/tutorials/0_generating_noise.ipynb [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "10df1455",
+   "metadata": {},
+   "source": [
+    "Finally, the generation of data for multiple detectors is supported by specifying the detector names as a \n",
+    "comma-separated string in the `detectors` key word. The list of supported detectors includes, amongst others,\n",
+    "the advanced LIGO detectors (`H1` for LIGO Hanford, `L1` for LIGO Livingston) \n",
+    "and the advanced Virgo detector `V1`."
```
Comment (dbkeitel) [inline examples/tutorials/0_generating_noise.ipynb:375]: Add comment on expected formats for `sqrtSX` and `timestamps` for multi-IFO?

File: examples/tutorials/0_generating_noise.ipynb [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93a86361",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Setup Writer\n",
+    "writer_kwargs ={\"label\": \"single_detector_gaussian_noise\",\n",
+    "                \"outdir\": \"Pyfstat_example_data\",\n",
```
[the remark this answers is not in the document]
Comment (dbkeitel) [reply examples/tutorials/0_generating_noise.ipynb:90]: and in all further example calls below, and also the signal notebook

File: examples/tutorials/0_generating_noise.ipynb [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+    "                \"tstart\": 1238166018,     # Starting time of the observation [GPS time]\n",
+    "                \"duration\": 5 * 86400,    # Duration [seconds]\n",
+    "                \"detectors\": \"H1\",        # Detector to simulate, in this case LIGO Hanford\n",
+    "                \"F0\": 100.,               # Central frequency band [Hz]\n",
+    "                \"Band\": 1.,               # Frequency band-width [Hz]\n",
```
[the remark this answers is not in the document]
Comment (dbkeitel) [reply examples/tutorials/0_generating_noise.ipynb:95]: and in all further example calls below

File: examples/tutorials/1_generating_signals.ipynb [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+    "The presence of visible CW features in the spectrogram can be quantified using the so-called\n",
+    "[sensitivity depth](https://arxiv.org/abs/1808.02459), which is defined as the ratio between the noise\n",
+    "and CW amplitude\n",
+    "$$\n",
+    "\\mathcal{D} = \\frac{\\sqrt{S_{\\text{n}}}}{h_0}\\;.\n",
+    "$$\n",
+    "\n",
+    "Current searches for CW signals from unknown sources are able to detect a sensitivity depth ranging between\n",
+    "$10\\;\\text{Hz}^{-1/2}$ and $50\\;\\text{Hz}^{-1/2}$. Visual features, however, tend to disappear around a depth\n",
+    "of $20\\;\\text{Hz}^{-1/2}$."
```
Comment (dbkeitel) [inline examples/tutorials/1_generating_signals.ipynb:186]: This sounds too optimistic for visual features according to my intuition, but I have problems running locally right now, will think about / test it again later.

**Review by dbkeitel — CHANGES_REQUESTED** [review]

File: examples/tutorials/1_generating_signals.ipynb [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+    "                \"tstart\": 1238166018,\n",
+    "                \"duration\": 365 * 86400,  \n",
+    "                \"detectors\": \"H1\",        \n",
+    "                \"sqrtSX\": 1e-23,          \n",
+    "                \"Tsft\": 1800,             \n",
+    "                \"SFTWindowType\": \"tukey\", \n",
+    "                \"SFTWindowBeta\": 0.01,\n",
+    "               }\n",
+    "\n",
+    "signal_parameters = pyfstat.AllSkyInjectionParametersGenerator(\n",
```
Comment (dbkeitel) [inline examples/tutorials/1_generating_signals.ipynb:306]: Can you rename this variable `signal_parameters_generator` or something, to make it easier to run cells in different order, and make it more explicit that this is something different than the simple dict before.

**Review by dbkeitel — APPROVED** [review]

**Pull request merged**